crypto.go

```diff
@@ -641,8 +843,26 @@ var aeads = map[AEADID]AEADScheme{
 	AEAD_CHACHA20POLY1305: chachaPolyScheme{},
 }
 
-func AssembleCipherSuite(kemID KEMID, kdfID KDFID, aeadID AEADID) (CipherSuite, error) {
-	kem, ok := kems[kemID]
+func AssembleKEMWithState(kemID KEMID, sk KEMPrivateKey, pk KEMPublicKey) (KEMScheme, bool) {
+	switch kemID {
+		case DHKEM_X25519: return dhkemScheme{group: x25519Scheme{}, skE: sk, pkE: pk}, true
+		case DHKEM_X448: return dhkemScheme{group: x448Scheme{}, skE: sk, pkE: pk}, true
+		case DHKEM_P256: return dhkemScheme{group: ecdhScheme{curve: elliptic.P256()}, skE: sk, pkE: pk}, true
+		case DHKEM_P521: return dhkemScheme{group: ecdhScheme{curve: elliptic.P521()}, skE: sk, pkE: pk}, true
+		case KEM_SIKE503: return sikeScheme{field: sidh.Fp503}, true
+		case KEM_SIKE751: return sikeScheme{field: sidh.Fp751}, true
+	}
+	return nil, false
+}
+
+func AssembleCipherSuite(kemID KEMID, kdfID KDFID, aeadID AEADID, sk KEMPrivateKey, pk KEMPublicKey) (CipherSuite, error) {
```

Comment:
I would revert this method, and make a duplicate, module-private one if you want.  (Or just repurpose the above method so that it assembles the whole suite instead of just the KEM.  We shouldn't be exposing our test points to the world.

Reply:
Yep, I'll revert and make the ephemeral-setter one module-private.